CMakeLists.txt

```
@@ -0,0 +1,31 @@
+set(WORMHOLE_WILLIAM_LIB
+    ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libwormhole_william.so)
+
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Windows")
+  set(EXPORT set)
+else()
+  set(EXPORT export)
+endif()
+
+add_custom_target(
+  BUILD_GO_LIBRARY ALL
+  COMMENT Building go static library
+  COMMAND ${EXPORT} CGO_ENABLED=1&&
+  	  ${EXPORT} GOARCH=${GOARCH}&&
+	  ${EXPORT} GOOS=${GOOS}&&
+	  ${EXPORT} CC_FOR_TARGET=${CC_FOR_TARGET}&&
+	  ${EXPORT} CC=${CC_FOR_TARGET}&&
+	  go build -buildmode=c-shared -o ${WORMHOLE_WILLIAM_LIB} ./c/client.c.go
```

I think we will need to include the whole package so that we can build more than that one file.

I think this should do it but haven't tested yet:
```suggestion
	  go build -buildmode=c-shared -o ${WORMHOLE_WILLIAM_LIB} ./c
```